beetsplug/kodinfo.py

```diff
@@ -0,0 +1,495 @@
+            position.text = str(album_tracks(albumid)[i][0])
+            title = et.SubElement(track, 'title')
+            title.text = album_tracks(albumid)[i][3]
+            duration = et.SubElement(track, 'duration')
+            duration.text = album_tracks(albumid)[i][1]
+            musicbrainztrackid = et.SubElement(track, 'musicBrainzTrackID')
+            musicbrainztrackid.text = album_tracks(albumid)[i][2]
+
+        xml = et.tostring(
+            root,
+            pretty_print=True,
+            xml_declaration=True,
+            encoding='UTF-8',
+            standalone="yes").decode()
+        print(xml, file=open(albumnfo, 'w+'))
+
+
+def artist_nfo_xml(albumid):
+    """Create XML file with artist information."""
+    for album in lib.albums(albumid):
+        artistid = 'mb_albumartistid:' + album.mb_albumartistid
+        artistnfo = os.path.join(artist_path(albumid)[0], 'artist.nfo')
+        if album.albumartist in ['Various Artists', 'Soundtracks',
+                                 'Compilations']:
+            pass
+        else:
+            root = et.Element('artist')
+            for i in range(len(artist_tags)):
+                artist_tags[i] = et.SubElement(
+                    root, '{}'.format(artist_tags[i]))
+                artist_tags[i].text = artist_info(albumid)[i]
+
+            for i in range(len(artist_path(albumid))):
+                path = et.SubElement(root, 'path')
+                path.text = artist_path(albumid)[i]
+
+            if artist_info(albumid)[11] == '':
+                thumb = et.SubElement(root, 'thumb')
+                thumb.text = ''
+            else:
+                thumb_location = os.path.join(
+                    album.path.decode('utf8'),
+                    os.pardir, 'artist.tbn')
+                urllib.request.urlretrieve(
+                    artist_info(albumid)[11], thumb_location)
+                thumb = et.SubElement(root, 'thumb')
+                thumb.text = artist_info(albumid)[11]
+                for i in range(len(thumbs('artist', albumid))):
+                    thumb = et.SubElement(root, 'thumb')
+                    thumb.text = thumbs('artist', albumid)[i]
+
+            fanart = et.SubElement(root, 'fanart')
+            fanart.text = artist_info(albumid)[12]
+
+            for i in range(len(artist_albums(artistid))):
+                album = et.SubElement(root, 'album')
+                title = et.SubElement(album, 'title')
+                title.text = artist_albums(artistid)[i][1]
+                year = et.SubElement(album, 'year')
+                year.text = str(artist_albums(artistid)[i][0])
+
+            xml = et.tostring(
+                root,
+                pretty_print=True,
+                xml_declaration=True,
+                encoding='UTF-8',
+                standalone="yes").decode()
+            print(xml, file=open(artistnfo, 'w+'))
+
+
+def update_kodi(host, port, user, password):
```

Comment:
Hmm… we already have a separate plugin to update the Kodi library (`kodiupdate`). Can we leave that responsibility to that plugin and just take care of the *.nfo file generation in this plugin?